### ISSUE TITLE
EOF occurred in violation of protocol (_ssl.c:590)

### ISSUE THREAD

**Issue by meisemma** [jay0lee/GAM#357]
Hi this is similar to these posts on the discussion wiki:
https://groups.google.com/forum/#!searchin/google-apps-manager/EOF%7Csort:relevance/google-apps-manager/ptUg9muhJg8/-nmtV825CAAJ

I am using GAM 4.03 for Windows. I'm  updating Google Classroom with adding a teacher to a course and also adding students to a course.  The operation does work but it is intermittent.  When doing this:
gam course Y12-Study-Skills sync students file Year12.csv
It failed 3 times with the EOF occurred in violation of protocol (_ssl.c:590) error.
Then on the fourth attempt it added 143 of 157 users as it skipped some of them with 
EOF occurred in violation of protocol (_ssl.c:590)
Then I tried it again, on the third attempt it added another 11 users.  
Now I'm trying it again, and I'm on the 7th attempt.
I've tried it behind a firewall and not behind a firewall, the same intermittent working and not working happen. 

Please can you help. I think GAM is brilliant and have used it for several years now.

Other people have issues on the Google group.  

https://groups.google.com/forum/#!searchin/google-apps-manager/EOF%7Csort:relevance/google-apps-manager/3ZkdlPXOmBw/19EJZo0RCgAJ

Thanks

**Comment by jay0lee** [comment]
These errors may indicate an issue with your network firewall / proxy / router. Can you also try creating a file called:

nocache.txt

in the exact same folder as gam.exe and see if that helps?

Jay

**Comment by meisemma** [comment]
Thank you :)

I tried that with nocache.txt.  I do believe it's our firewall as I tethered it to my phone connection - worked ok. And as soon as I reconnected it to the LAN same EOF.

I've read that you support proxy. I entered the set http_proxy details with our IP address, but it returned an error. But we are on a direct connection anyway. So I don't think I need to :) ??

I think it's more a case that our ISP or router/firewall doesn't have the correct certificate or is having problems with googleapis.com.  I've tried searching in Google's documentation. I know you just connect straight through to http/https at Google's servers so there's no IPs to allow for GAM itself.

Is there a specific test I can do to test we are connecting to Google's servers probably? Or page I can go to? Problem is that it's an intermittent connection problem, something is timing out somewhere. Ping seems to work etc.
